src/Endpoint/PartitionProviderInterface.php

```diff
@@ -0,0 +1,14 @@
+<?php
+namespace Aws\Endpoint;
+
+interface PartitionProviderInterface
+{
+    /**
+     * Gets the partition containing a given region.
+     * 
+     * @param string $region
+     *
+     * @return Partition
+     */
+    public function getPartition($region);
```

Comment:
In order to allow this interface to be more extensible, perhaps this should be a more targeted method name... Like getPartitionFromRegion($region) or something.